### PR TITLE
@keys: Ensure single grapheme cluster and unicode normalization

### PR DESCRIPTION
**Pull request by LeonMatthes**
- **Add Unicode normalization and grapheme cluster validation for keyboard shortcuts**
- **Add string-based keyboard testing API and Unicode normalization test**

As discussed in #102, this ensures consistent handling for different languages.
Especially the unicode normalization is crucial to ensure keyboard shortcuts involving accents or other composed characters can be matched consistently.

### CODE REVIEW

File: internal/core/Cargo.toml [hunk 1]
```diff
@@ -48,7 +48,7 @@ std = [
 # from a single core, and not in a interrupt or signal handler.
 unsafe-single-threaded = []
 
-unicode = ["unicode-script", "unicode-linebreak"]
+unicode = ["unicode-script", "unicode-linebreak", "unicode-normalization"]
```
Comment (tronical) [inline internal/core/Cargo.toml]: I wonder if this really works....

I can see `unicode` in `default`, but we're not using `default` when building a Rust Slint application. Perhaps this should be in `std`?

Comment (LeonMatthes) [reply internal/core/Cargo.toml]: Ah, I thought the `slint` crate pulled in `unicode` by default. Weird that we disable default-features on i-slint-core. Then what's the point of having default features there at all?

Why is the `unicode` feature not enabled by default anyway? We want our default features to give a good out-of-the-box experience, right? For international users, enabling enabled should be the default IMO :thinking:

Comment (LeonMatthes) [reply internal/core/Cargo.toml]: Note: I'm digging into this a bit more, and it seems that all these crates enable the `default` feature set:

* `interpreter`
* `femtovg`
* `skia`
* `linuxkms`
* `winit`
* `node`

Which causes `unicode` and the other `default` features to be essentialy default-enabled on desktop.

To sort this out I think the best way:

* `unicode` cannot be enabled via the `slint` package - we should probably add a `unicode` feature to the Slint API crate
* Remove the current `default` features of `i-slint-core` and move them into a `desktop` feature. Disabling the `default` features by default seems backwards...

Comment (LeonMatthes) [reply internal/core/Cargo.toml]: > * Remove the current `default` features of `i-slint-core` and move them into a `desktop` feature. Disabling the `default` features by default seems backwards...

Ah, this is the case for all our internal workspace packages :see_no_evil: 
So for now let's stick with this, but just enable `unicode` by default and allow users to control this by exposing it as a public feature on the `slint` crate

**Review by tronical — APPROVED** [review]
